src/Xamarin.Android.Tools.AndroidSdk/AndroidSdkInfo.cs

```diff
@@ -155,12 +155,12 @@ internal static void DefaultConsoleLogger (TraceLevel level, string message)
 			}
 		}
 
-		public static void SetPreferredAndroidSdkPath (string path, Action<TraceLevel, string> logger = null)
+		public static bool SetPreferredAndroidSdkPath (string path, Action<TraceLevel, string> logger = null)
```

Comment:
This is "public"-ish API, used in numerous places, and it might not be easy to keep all such places synchronized with each other.  This should thus be treated as an ABI break and avoided if at all possible.

Additionally, I fail to see why it would be a "bad" thing for `AndroidSdkInfo.SetPreferredAndroidSdkPath()` to throw an exception when it can't save the specified value.  Why can't the *caller* catch the exception?

Reply:
This issue is caused because user do not have the right permissions to create the `$HOME/.config/xbuild/monodroid-config.xml` and it is caused by https://github.com/xamarin/xamarin-android-tools/pull/71/commits/4dca19a818ff7931240e2165ebb8002fab52031c#diff-c96432cf25d2723c79c03d426742f345L192.  The issue is coming from xamarin-android-tools which is a submodule for installer.  My fix is to handle this issue properly in android-tools, so that installer can handle it  with an error slide and the user knows that installer do not have the right permissions to create $HOME/.config/xbuild/monodroid-config.xml. This is not hiding the error but a way to show user that  there is a permission issue and installer cannot continue Xamarin.Android installation.